### PR TITLE
Generate streaming endpoints that specify stream-condition

### PR DESCRIPTION
**Pull request by amckinney**
This extends the `x-fern-streaming` OpenAPI extension syntax so that it lets users define an endpoint that has both a unary and streaming version of the same method, which is distinguished with a boolean property set to true or false, which is beginning to be a common usage pattern seen in several (optionally) streaming APIs.

Example usage is shown below:

```yaml
paths:
  /v1/users:
    post:
      operationId: user
      requestBody:
        content:
          application/json:
            schema:
              type: object
              properties:
                id:
                  description: ""
                  type: string
      x-fern-streaming:
        stream-condition: $request.stream
        response:
          $ref: "#/components/schemas/User"
        response-stream:
          $ref: "#/components/schemas/StreamUser"
      responses:
        "200":
          description: ""
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/User"
components:
  schemas:
    User:
      type: object
      properties:
        id:
          type: string
          format: uuid
    StreamUser:
      type: object
      properties:
        id:
          type: string
          format: uuid
```

### CODE REVIEW

File: packages/cli/openapi-parser/src/v3/converters/convertSchemas.ts [hunk 1]
```diff
@@ -148,6 +148,13 @@ export function convertSchemaObject(
 
     // primitive types
     if (schema === "boolean" || schema.type === "boolean") {
+        const literalValue = getExtension<boolean>(schema, FernOpenAPIExtension.BOOLEAN_LITERAL);
+        if (literalValue !== undefined) {
+            return SchemaWithExample.literal({
+                value: LiteralSchemaValue.boolean(literalValue),
+                description,
+            });
```
Comment (dsinghvi) [inline packages/cli/openapi-parser/src/v3/converters/convertSchemas.ts:156]: we're going to want to call something like `wrapPrimitive`

Comment (dsinghvi) [reply packages/cli/openapi-parser/src/v3/converters/convertSchemas.ts:156]: `wrapPrimitive` handles wrapping in nulllable/optional depending on what has been specified

File: packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts [hunk 2]
```diff
@@ -31,10 +35,191 @@ export function convertStreamingOperation({
...
+                    ...operationContext,
+                    operation: {
+                        ...operationContext.operation,
+                        requestBody: convertRequestBodyToInlinedRequest({
+                            context,
+                            requestBody: operationContext.operation.requestBody,
+                            streamCondition: streamingExtension,
+                            isStreaming: true,
+                        }),
+                    },
```
Comment (dsinghvi) [inline packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts:50]: you're probably already tracking but you'll have to also create a similar helper `createOperationResponse(streamingExtension.responseStream)`

Comment (dsinghvi) [reply packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts:50]: basically this helper will return 
```ts
{
  "200": {
    "application/json": {
        "schema": responseSchema
    }
  }
}
```

Comment (amckinney) [reply packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts:50]: Yup, tracking that but thanks for calling that out - need to port the implementation over from what we have in https://github.com/fern-api/fern/pull/2247

File: packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts [hunk 3]
```diff
@@ -31,10 +35,191 @@ export function convertStreamingOperation({
...
         }
-        case "streamCondition":
-            break;
+        case "streamCondition": {
+            const streamingOperation = convertHttpOperation({
+                operationContext: {
+                    ...operationContext,
+                    operation: {
+                        ...operationContext.operation,
+                        requestBody: convertRequestBodyToInlinedRequest({
```
Comment (dsinghvi) [inline packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts:44]: thoughts on renaming this to `createStreamingRequestBody({ context, requestBody, streamConditionProperty })`; 

`inlinedRequest` is more a fern term so figured we could remove that

Comment (amckinney) [reply packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts:44]: Totally fine with that, let's do it

File: packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts [hunk 4]
```diff
@@ -31,10 +35,191 @@ export function convertStreamingOperation({
...
+    requestBody,
+    streamCondition,
+    isStreaming,
+}: {
+    context: AbstractOpenAPIV3ParserContext;
+    requestBody?: OpenAPIV3.ReferenceObject | OpenAPIV3.RequestBodyObject;
+    streamCondition: StreamConditionEndpoint;
+    isStreaming: boolean;
+}): OpenAPIV3.RequestBodyObject {
+    if (requestBody === undefined) {
```
Comment (dsinghvi) [inline packages/cli/openapi-parser/src/v3/converters/operation/convertStreamingOperation.ts:93]: `== null`

File: packages/cli/openapi-ir-to-fern/src/__test__/__snapshots__/streaming.test.ts.snap [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+            "request": {
+              "body": {
+                "properties": {
+                  "id": {
+                    "docs": "",
+                    "type": "optional<string>",
+                  },
+                  "stream": {
+                    "docs": undefined,
+                    "type": "optional<literal<false>>",
```
Comment (amckinney) [inline packages/cli/openapi-ir-to-fern/src/__test__/__snapshots__/streaming.test.ts.snap:40]: This shouldn't be optional - the literal we synthesize from `x-fern-streaming` should be sent unconditionally.

Comment (amckinney) [reply packages/cli/openapi-ir-to-fern/src/__test__/__snapshots__/streaming.test.ts.snap]: Done in https://github.com/fern-api/fern/pull/2251/commits/1b763b2f791a071504c9f9a7aa05a668f39400f8

**Review by dsinghvi — APPROVED** [review]

**Pull request merged**